common/kotlinx-coroutines-core-common/src/JobSupport.kt

```diff
@@ -1054,7 +1059,7 @@ public open class JobSupport constructor(active: Boolean) : Job, ChildJob, Paren
     public val isCompletedExceptionally: Boolean get() = state is CompletedExceptionally
 
     public fun getCompletionExceptionOrNull(): Throwable? {
-        val state = this.state
+        val state = this.state.unboxState()
```

Comment:
Unboxing is not needed and even harmful here. This method would not work problem with boxed incomplete state.